src/common_types.ts

```diff
@@ -194,33 +194,51 @@ export type PluginOutput = {
 export type ScreenshotOptions = 'on' | 'off' | 'only-on-failure';
 
 export type CliArgs = {
-  capability?: Array<string>;
   config?: string;
-  outfd?: number;
+  params?: Params;
+  reporter?: Reporters;
+  pattern?: string;
+  inline?: boolean;
+  require?: Array<string>;
   headless?: boolean;
+  sandbox?: boolean;
+  richEvents?: boolean;
+  capability?: Array<string>;
   screenshots?: ScreenshotOptions;
-  ssblocks?: boolean;
-  metrics?: boolean;
-  filmstrips?: boolean;
-  trace?: boolean;
   dryRun?: boolean;
-  network?: boolean;
-  pauseOnError?: boolean;
-  quietExitCode?: boolean;
-  reporter?: Reporters;
-  wsEndpoint?: string;
-  sandbox?: boolean;
-  pattern?: string;
-  inline: boolean;
   match?: string;
   tags?: Array<string>;
-  require: Array<string>;
-  debug?: boolean;
+  outfd?: number;
+  wsEndpoint?: string;
+  pauseOnError?: boolean;
   ignoreHttpsErrors?: boolean;
-  params?: Params;
   throttling?: boolean | string;
-  playwrightOptions?: LaunchOptions & BrowserContextOptions;
-  richEvents?: boolean;
+  playwrightOptions?: PlaywrightOptions;
+};
+
+export type RunOptions = Omit<
+  CliArgs,
+  | 'config'
+  | 'pattern'
+  | 'inline'
+  | 'require'
+  | 'reporter'
+  | 'richEvents'
+  | 'capability'
+  | 'sandbox'
+  | 'headless'
+  | 'throttling'
+> & {
+  quietExitCode?: boolean;
+  metrics?: boolean;
+  ssblocks?: boolean;
+  network?: boolean;
+  trace?: boolean;
+  filmstrips?: boolean;
+  environment?: string;
+  playwrightOptions?: PlaywrightOptions;
+  networkConditions?: NetworkConditions;
+  reporter?: CliArgs['reporter'] | Reporter;
 };
```

Comment:
Nit: I'd probably have a `BaseArgs` type which I'd then only add to with intersection types, so you'd have something like:

```ts
type BaseArgs = {
  params?: Params;
  screenshots?: ScreenshotOptions;
  dryRun?: boolean;
  match?: string;
  tags?: Array<string>;
  outfd?: number;
  wsEndpoint?: string;
  pauseOnError?: boolean;
  ignoreHttpsErrors?: boolean;
  playwrightOptions?: PlaywrightOptions;
}

type CliArgs = BaseArgs & {
  config?: string;
  reporter?: Reporters;
  pattern?: string;
  inline?: boolean;
  require?: Array<string>;
  headless?: boolean;
  sandbox?: boolean;
  richEvents?: boolean;
  capability?: Array<string>;
  throttling?: boolean | string;
}

type RunOptions = BaseArgs & { 
  quietExitCode?: boolean;
  metrics?: boolean;
  ssblocks?: boolean;
  network?: boolean;
  trace?: boolean;
  filmstrips?: boolean;
  environment?: string;
  playwrightOptions?: PlaywrightOptions;
  networkConditions?: NetworkConditions;
  reporter?: CliArgs['reporter'] | Reporter;
}
```

IMO that makes it easier to understand what are the commonalities between the options for different commands.

Reply:
Good point ++